### PR TITLE
Feature/import warnings

### PR DESCRIPTION
**Pull request by Stevenjin8**
Fix for https://github.com/Azure/azure-sdk-for-python/issues/24989.

### CODE REVIEW

File: tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py [hunk 1]
```diff
@@ -1914,7 +1914,46 @@ def visit_module(self, node):
...
+
+class BlockedImport(BaseChecker):
+    """There are certain imports that should only occur in the core package.
+    For example, instead of using `requests` to make requests, clients should
+    take a `azure.core.pipeline.Pipeline` as input to make requests.
+    """
+    name = "blocked-import"
+    priority = -1  # what is this?
+    msgs = {
+        "W9992": (
```
Comment (l0lawrence) [inline tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py:1926]: I think we want this to follow the other checkers and be C# for Convention

File: tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py [hunk 2]
```diff
@@ -1914,7 +1914,46 @@ def visit_module(self, node):
...
                 logger.debug("Pylint custom checker failed to check if model is aliased.")
-                pass
+
+class BlockedImport(BaseChecker):
+    """There are certain imports that should only occur in the core package.
+    For example, instead of using `requests` to make requests, clients should
+    take a `azure.core.pipeline.Pipeline` as input to make requests.
+    """
+    name = "blocked-import"
+    priority = -1  # what is this?
```
Comment (l0lawrence) [inline tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py:1924]: priority is " This must be to be lower than 0. The checkers are ordered by
the priority when run, from the most negative to the most positive." - pylint

**Comment by lmazuel** [comment]
General comment: this should apply to everybody, even azure-core. The only files allowed to import requests are the transport of requests (IOW, files like paging, lro, exceptions, etc. of azure-core are not waived of this check). Same for aiohttp.

Of we do a global suppression, I would do only azure/core/pipeline/transport, and no other folders.

**Review by tjprescott — COMMENTED** [review]
This looks good to me. Have you run this on the repo to see how many violations exist?

**Comment by Stevenjin8** [comment]
@tjprescott Running it right now, might take a sec

**Comment by Stevenjin8** [comment]
@tjprescott I tried running pylint locally, but my computer ran out of memory. Instead, I just searched patterns like `import requests\.`, `import aiohttp\.`, `from aiohttp* import` in Vscode. There were around 50 violations.

**Comment by tjprescott** [comment]
> @tjprescott I tried running pylint locally, but my computer ran out of memory. Instead, I just searched patterns like import requests\., import aiohttp\., from aiohttp* import in Vscode. There were around 50 violations. Not sure how to proceed.
> 
> EDIT: I'll also add that a lot of them are unused imports

Yikes. Perhaps we need to focus the logic? Running out of memory won't do us any good.

**Comment by Stevenjin8** [comment]
> > @tjprescott I tried running pylint locally, but my computer ran out of memory. Instead, I just searched patterns like import requests., import aiohttp., from aiohttp* import in Vscode. There were around 50 violations. Not sure how to proceed.
> > EDIT: I'll also add that a lot of them are unused imports
> 
> Yikes. Perhaps we need to focus the logic? Running out of memory won't do us any good.

@tjprescott I'm not sure what you mean by focusing the logic. Do you mean the code for the linter? or just how I'm running them?

My hypothesis for the memory problems was that pylint checks for code similarity, but it still got stuck when I tried to disable code similarity. I actually tried disabling all linter rules except for mine. I dont remember exactly what I ran, but it also started to hog memory.

I'm running the linters in WSL... maybe that's contributing?  `htop` tells me I have 24 gigs of memory, so that doesn't seem to be the problem.

**Comment by scbedd** [comment]
@Stevenjin8 I noticed that `tools - asset-sync` blew up on you. That is totally unexpected and SHOULD NOT be blocking you. I have a PR [over here](https://github.com/Azure/azure-sdk-tools/pull/3776) that remediates the errors you see here. Once you get pylint-guidelines-checker green you can ignore the asset-sync failures with `/check-enforcer override` if you need to!

**Review by tjprescott — APPROVED** [review]

**Comment by Stevenjin8** [comment]
I managed to get pylint to run on a vm (only used 35 gigs of memory!) I disabled all rules except for this one:

```
************* Module appconfiguration.azure-appconfiguration.azure.appconfiguration.aio._azure_configuration_client_async
sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_configuration_client_async.py:8:0: C4749: This import is not allowed here. Consider using an abstract alternative from azure.core.pipeline.transport. (networking-import-outside-azure-core-transport)
************* Module core.azure-servicemanagement-legacy.azure.servicemanagement._http.httpclient
sdk/core/azure-servicemanagement-legacy/azure/servicemanagement/_http/httpclient.py:107:12: C4749: This import is not allowed here. Consider using an abstract alternative from azure.core.pipeline.transport. (networking-import-outside-azure-core-transport)
************* Module core.azure-servicemanagement-legacy.azure.servicemanagement.servicemanagementclient
sdk/core/azure-servicemanagement-legacy/azure/servicemanagement/servicemanagementclient.py:19:0: C4749: This import is not allowed here. Consider using an abstract alternative from azure.core.pipeline.transport. (networking-import-outside-azure-core-transport)
************* Module core.azure-core.azure.core.pipeline.transport._aiohttp
sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py:35:0: C4749: This import is not allowed here. Consider using an abstract alternative from azure.core.pipeline.transport. (networking-import-outside-azure-core-transport)
************* Module core.azure-core.azure.core.pipeline.transport._requests_basic
sdk/core/azure-core/azure/core/pipeline/transport/_requests_basic.py:34:0: C4749: This import is not allowed here. Consider using an abstract alternative from azure.core.pipeline.transport. (networking-import-outside-azure-core-transport)
************* Module core.azure-core.azure.core.pipeline.transport._bigger_block_size_http_adapters
sdk/core/azure-core/azure/core/pipeline/transport/_bigger_block_size_http_adapters.py:28:0: C4749: This import is not allowed here. Consider using an abstract alternative from azure.core.pipeline.transport. (networking-import-outside-azure-core-transport)
************* Module core.azure-core.azure.core.pipeline.transport._requests_trio
sdk/core/azure-core/azure/core/pipeline/transport/_requests_trio.py:32:0: C4749: This import is not allowed here. Consider using an abstract alternative from azure.core.pipeline.transport. (networking-import-outside-azure-core-transport)
sdk/core/azure-core/azure/core/pipeline/transport/_requests_trio.py:35:0: C4749: This import is not allowed here. Consider using an abstract alternative from azure.core.pipeline.transport. (networking-import-outside-azure-core-transport)
************* Module core.azure-core.azure.core.pipeline.transport._requests_asyncio
sdk/core/azure-core/azure/core/pipeline/transport/_requests_asyncio.py:35:0: C4749: This import is not allowed here. Consider using an abstract alternative from azure.core.pipeline.transport. (networking-import-outside-azure-core-transport)
************* Module core.azure-core.azure.core.rest._requests_basic
sdk/core/azure-core/azure/core/rest/_requests_basic.py:31:0: C4749: This import is not allowed here. Consider using an abstract alternative from azure.core.pipeline.transport. (networking-import-outside-azure-core-transport)
************* Module core.azure-core.azure.core.rest._requests_trio
sdk/core/azure-core/azure/core/rest/_requests_trio.py:26:0: C4749: This import is not allowed here. Consider using an abstract alternative from azure.core.pipeline.transport. (networking-import-outside-azure-core-transport)
************* Module batch.azure-batch.azure.batch.batch_auth
sdk/batch/azure-batch/azure/batch/batch_auth.py:15:0: C4749: This import is not allowed here. Consider using an abstract alternative from azure.core.pipeline.transport. (networking-import-outside-azure-core-transport)
sdk/batch/azure-batch/azure/batch/batch_auth.py:16:0: C4749: This import is not allowed here. Consider using an abstract alternative from azure.core.pipeline.transport. (networking-import-outside-azure-core-transport)
************* Module storage.azure-storage-blob.azure.storage.blob.aio._download_async
sdk/storage/azure-storage-blob/azure/storage/blob/aio/_download_async.py:15:0: C4749: This import is not allowed here. Consider using an abstract alternative from azure.core.pipeline.transport. (networking-import-outside-azure-core-transport)
------------------------------------
Your code has been rated at 10.00/10
```

p.s. I did take down the vm

**Comment by Stevenjin8** [comment]
I realize that its warning about azure-core module, but I think this is because I ran pylint from the root of the repo

**Comment by Stevenjin8** [comment]
for `azure-core`, I get

```
************* Module azure.core.rest._requests_basic
azure/core/rest/_requests_basic.py:31:0: C4749: This import is not allowed here. Consider using an abstract alternative from azure.core.pipeline.transport. (networking-import-outside-azure-core-transport)
************* Module azure.core.rest._requests_trio
azure/core/rest/_requests_trio.py:26:0: C4749: This import is not allowed here. Consider using an abstract alternative from azure.core.pipeline.transport. (networking-import-outside-azure-core-transport)
```

**Review by l0lawrence — APPROVED** [review]
We might want to bump the version of the pylint-checker plugin as well since we are adding a new checker to .0.0.7

**Pull request merged**